src/engine/pieces/bishop.ts

```diff
@@ -1,13 +1,24 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from '../square';
 
 export default class Bishop extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let availableMoves = new Array<Square>();
+        let currentPosition = board.findPiece(this);
+        
+        this.pathCheck(-1,-1,availableMoves,currentPosition,board);
+        this.pathCheck(-1,1,availableMoves,currentPosition,board);
+        this.pathCheck(1,-1,availableMoves,currentPosition,board);
+        this.pathCheck(1,1,availableMoves,currentPosition,board);
```

Comment:
could simplify this with a `Steps` object that looks a bit like
`[[0,1], ... [-1,0]]` then you loop through calling `this.pathCheck` on that

Also it's common practice to have spaces between values passing in to a function

Reply:
Fixed